src/Paprika/Store/StorageFanOutPage.cs

```diff
@@ -107,18 +107,31 @@ public void Destroy(IBatchContext batch, in NibblePath prefix)
         // Destroy the Id entry about it
         Set(prefix, ReadOnlySpan<byte>.Empty, batch);
 
-        // Destroy the account entry
-        // SetAtRoot<FanOutPage>(batch, account, ReadOnlySpan<byte>.Empty, ref Data.StateRoot);
-
-        // Remove the cached
-        batch.IdCache.Remove(prefix.UnsafeAsKeccak);
-        var index = GetIndex(prefix);
-        var addr = Data.Addresses[index];
-        if (addr.IsNull)
+        var map = new SlottedArray(Data.Data);
+        foreach (var item in map.EnumerateAll())
         {
-            // recycleForReuse
+            if (item.Key.Equals(prefix))
+            {
+                map.Delete(item);
+                var sliced = prefix.SliceFrom(ConsumedNibbles);
```

Comment:
This loop over indexes should be separate. Otherwise you'll perform the check for each item.  Conceptually, I'd follow:

1. scan map and delete what is needed
2. propagate down

The first, could be moved to the map as discussed before so that this is a single call. The 2nd, it's specific for a page type.

Reply:
Pls check now.